contracts/adapters/IPRebalanceExtension.sol

```diff
@@ -0,0 +1,478 @@
+     * TransformHelper.
+     *
+     * @param _transformComponent       component to transform
+     * @param _transformData            transform data
+     */
+    function transformRemaining(
+        address _transformComponent,
+        bytes memory _transformData
+    )
+        external
+        onlyAllowedCaller(msg.sender)
+    {
+        address underlying = transformComponentInfo[_transformComponent].underlyingComponent;
+        uint256 underlyingUnitsRaw = rebalanceParams[underlying];
+        require(underlyingUnitsRaw == 0, "raw underlying in target set composition");
+
+        _transform(_transformComponent, _transformData, true);
+    }
+
+    /* ======== Internal Functions ======== */
+
+    /**
+     * Untransforms a component. If it is the final untransform, it will automatically begin the rebalance
+     * through GeneralIndexModule.
+     */
+    function _untransform(address _transformComponent, bytes memory _untransformData) internal {
+
+        TransformInfo memory transformInfo = transformComponentInfo[_transformComponent];
+
+        require(transformInfo.minTransformDelay + lastTransform[_transformComponent] < block.timestamp, "delay not elapsed");
+        lastTransform[_transformComponent] = block.timestamp;
+
+        require(transformInfo.underlyingComponent != address(0), "nothing to untransform");
+        require(
+            transformInfo.transformHelper.shouldUntransform(transformInfo.underlyingComponent, _transformComponent),
+            "untransform unavailable"
+        );
+
+        uint256 targetUnitsUnderlying = rebalanceParams[_transformComponent];
+        uint256 currentUnits = setToken.getDefaultPositionRealUnit(_transformComponent).toUint256();
+
+        // convert target units from underlying to transformed amounts
+        uint256 exchangeRate = transformInfo.transformHelper.getExchangeRate(transformInfo.underlyingComponent, _transformComponent);
+        uint256 targetUnitsInTransformed = targetUnitsUnderlying.preciseMul(exchangeRate);
+        uint256 unitsToUntransform = currentUnits > targetUnitsInTransformed ? currentUnits.sub(targetUnitsInTransformed) : 0;
+        
+        uint256 maxUntransformUnits = transformInfo.maxTransformSize.preciseDiv(setToken.totalSupply()).preciseMul(exchangeRate);
+        if (unitsToUntransform > maxUntransformUnits) {
+            unitsToUntransform = maxUntransformUnits;
+        }
+
+
+        require(unitsToUntransform > 0, "nothing to untransform");
+
+        (address module, bytes memory callData) = transformInfo.transformHelper.getUntransformCall(
+            manager.setToken(),
+            transformInfo.underlyingComponent,
+            _transformComponent,
+            unitsToUntransform,
+            _untransformData
+        );
+
+        invokeManager(module, callData);
+    }
+
+    /**
+     * Untransforms a component
```

Comment:
What is this function actually supposed to do? What is the math that should be happening? This javadoc tells me nothing about how this function is supposed to work